server/ai_process.go

```diff
@@ -81,19 +85,34 @@ func processTextToImage(ctx context.Context, params aiRequestParams, req worker.
 		return nil, err
 	}
 
-	imgResp := resp.(*worker.ImageResponse)
+	imgResp, ok := resp.(*worker.ImageResponse)
+	if !ok {
+		return nil, errWrongFormat
+	}
 
 	newMedia := make([]worker.Media, len(imgResp.Images))
 	for i, media := range imgResp.Images {
+		var result []byte
 		var data bytes.Buffer
+		var name string
 		writer := bufio.NewWriter(&data)
-		if err := worker.ReadImageB64DataUrl(media.Url, writer); err != nil {
-			return nil, err
+		err := worker.ReadImageB64DataUrl(media.Url, writer)
+		if err == nil {
```

Comment:
How is this change related to Remote Worker? I can't connect the dots.
Have we started sending now the image in the URL instead of Base64encoded?

Reply:
The ai-worker runner returns base64 encoded results in the `.Url`field.  In the implementation now the orchestrator recieved the base64 encoded results and sends it up to the gateway to decode and save in NodeStorage session.

I thought passing base64 encoded results from remote ai worker > orchestrator > gateway was not ideal so I updated to using similar implementation as transcoding that uses node storage and passes urls.
 
This is part of the backwards compatibility so gateways can process both results from orchestrators that have not updated and orchestrators using the updated implementation.  Some of this can be removed after a period of time to let orchestrators update.  Or could just force the update on to orchestrators to enable not needing some of this.

Reply:
Ok, I see. It makes sense. 

Tagging @thomshutt , because it's somehow related to the issue with returning large data from SAM2 (I know your team is working on that).

Reply:
The ai-worker decodes the base64 result from the ai-runner and then posts a multipart result to the orchestrator `/aiResults` endpoint.  For binary results (image and video), the orchestrator then saves the result in node storage and provides a url for the gateway to download the result from.  This is the workflow that transcoding uses and I thought would be good to start to separate out the result when it is binary.  JSON responses from `audio-to-text` and `segment-anything-2` pass through without this intermediate step to split out the result since they are not base64 encoded.  With the experience from SAM2 and the large responses, wondering if all pipelines should separate the result from the response to the request to generate.  Thoughts?

Reply:
IMO it would nice to have the same flow for all models. And I believe this does not add much to the overall latency. @thomshutt Do you have any opinion on that?

Reply:
Agree, I think that splitting out the response from the reponse data to aid in dealing with large responses should probably come in its own PR and should apply to all models for consistency.

Reply:
Ok, I agree. To confirm, don't undo what I already have. Just punt on adding things like SSE to other pipelines to a new PR right? Or saying put the text based responses in URLs to download as well?  I have reservations doing this on the LLM pipeline to be honest.

Reply:
I think it's ok to do what you suggested, meaning "don't undo what you have".